docs/understanding-radicle/how-it-works.md

```diff
@@ -1,80 +1,488 @@
+
+The `CodeSigningKey` used in verification may be equal to the `DeviceKey` for
+any device `d` (i.e. as long as the key is not shared between multiple devices).
+Typically, PGP keys already present on a PKI are used for code signing, and
+shared across devices.
+
+*See more on [key architecture][sc]*
+
+
+### Verification
+
+Document revisions, as well as authenticity attestations, form a hash-linked
+chain leading to the initial revision of the document. In order to verify the
+authenticity of a given identity attestation (`Identity`), the chain of updates
+must be contiguous and subject to the following verification rules.
+
+There are four levels of validity:
+
+1. **Untrusted**
+
+    The identity document is well-formed, and points to a root object which is
+    retrievable from local storage.
+
+2. **Signed**
+
+   The identity carrier passes **1.**, and is signed by at least one key
+   specified in the delegations of the document.
+
+3. **Quorum**
+
+    The identity carrier passes **2.**, and is signed by a quorum of the keys
+    specified in the delegations of the document (`Q > D/2`).
+
+4. **Verified**
+
+    The identity carrier passes **3.**, and: * The document does not refer to a
+        previous revision, and no previous revision is known * Or, the set of
+        signatures forms a quorum of the [delegations][dl] of the previous
+        revision.
+
+    It is an error if: * No previous revision is given, but a parent in the
+        identity attestation chain is found * A previous revision is given, but
+        the identity attestation chain does not yield a parent * A previous
+        revision is given, but it is not the same the parent attestation refers
+        to * The current and parent attestations refer to different roots
+
+*See more notes on [Verification][ve]*
+
+## Git Implementation
+
+### Overview
+
+Radicle basically uses Git as a database. This means everything is stored in
+single Git monorepo that is read and written from via the Upstream client. Our
+Git implementation was devised to create an incentive for the seeder to provide
+all data necessary to resolve and verify a repository, while reducing latency by
+eliminating gossip queries and git fetches as much as possible. 
+
+Read more about our approach [here][ir]
+
+### Namespacing
+
+Radicle uses [namespaces][ns] to store a `peer`'s Radicle data. By using
+namespacing, Radicle Link partitions a single git repository - or monorepo -
+into logical, smaller repos, that can be checked out individually. This monorepo
+includes a `peer`'s identity (personal & project) data, as well as all tracked
+Git repositories.
+
+The namespacing scheme has the following format:
+
+```rust
+# Owner of the monorepo
+let PEER_ID;
+
+# Peer tracked by $PEER_ID, either directly or transitively
+let TRACKED_PEER_ID;
+
+# Identity hash of the project or peer
+let IDENTITY;
+
+# Identity hashes of certifiers of $IDENTITY
+let CERTIFIER[1..];
```

Comment:
`s/CERTIFIER/DELEGATE/`

It is the personal identities a _project_ delegates to (nb. a _personal_ identity can only delegate to _keys_, not other identities).

Reply:
There is simply no `ids` then, because there is no link to follow.

Reply:
Oh ya good point :+1:

Reply:
So, what needs to be changed here?